clang/lib/Sema/SemaChecking.cpp

```diff
@@ -9827,6 +9827,9 @@ static IntRange GetExprRange(ASTContext &C, const Expr *E, unsigned MaxWidth,
     return IntRange(BitField->getBitWidthValue(C),
                     BitField->getType()->isUnsignedIntegerOrEnumerationType());
 
```

Comment:
@shafik Getting the expression range for a void type isn't handled at this stage

https://github.com/llvm/llvm-project/blob/bf1a554312bd011cb2016a2c9d7e75d6fe3b02af/clang/lib/Sema/SemaChecking.cpp#L9628-L9631

Throw expressions can be handled if `CondResult` isn't treated as a boolean

https://github.com/llvm/llvm-project/blob/bf1a554312bd011cb2016a2c9d7e75d6fe3b02af/clang/lib/Sema/SemaChecking.cpp#L9633-L9644

It seems this case could be handled either before

https://github.com/llvm/llvm-project/blob/bf1a554312bd011cb2016a2c9d7e75d6fe3b02af/clang/lib/Sema/SemaChecking.cpp#L9629-L9631

or by adding a condition in the recursion call to handle void types. Which approach do you think is better? Thanks

Reply:
What expr is ending up in `GetExprRange` b/c it looks like the `E = E->IgnoreParens()` should do the job? Right?

Reply:
The parentheses are omitted at the start of the method

 https://github.com/llvm/llvm-project/blob/ed3d05178274890fb804f43ae1bcdfd33b5fd8f0/clang/lib/Sema/SemaChecking.cpp#L9587

Reply:
So why doesn't this `if` statement handle it?

https://github.com/llvm/llvm-project/blob/ed3d05178274890fb804f43ae1bcdfd33b5fd8f0/clang/lib/Sema/SemaChecking.cpp#L9625

Is this line not handling it correctly? 

https://github.com/llvm/llvm-project/blob/ed3d05178274890fb804f43ae1bcdfd33b5fd8f0/clang/lib/Sema/SemaChecking.cpp#L9637

Reply:
At this point, we get

```
ConditionalOperator 0x1108658e0 'int'
|-CXXBoolLiteralExpr 0x110865878 '_Bool' true
|-CXXThrowExpr 0x1108658a8 'void'
| `-IntegerLiteral 0x110865888 'int' 0
`-IntegerLiteral 0x1108658c0 'int' 0
```

nd as a result, `GetExprRange` proceeds with the throw expression (`CO->getTrueExpr()`)

https://github.com/llvm/llvm-project/blob/ed3d05178274890fb804f43ae1bcdfd33b5fd8f0/clang/lib/Sema/SemaChecking.cpp#L9628-L9631

Reply:
I think that makes more sense now, can you add some of these details to the summary please.

Reply:
I've updated the PR details